lighthouse-core/lib/traces/metrics-evts.js

```diff
@@ -0,0 +1,204 @@
+          return ttiExt.value.timestamps.timeToInteractive;
+        }
+      }
+    ];
+  }
+
+  /**
+   * Returns simplified representation of all metrics' timestamps from monotonic clock
+   * @param {!Array<!AuditResult>} auditResults
+   * @return {!Array<{ts: number, id: string, name: string}>} metrics to consider
+   */
+  gatherMetrics(auditResults) {
+    const metricDfns = Metrics.metricsDefinitions;
+    const resolvedMetrics = [];
+    metricDfns.forEach(metric => {
+      // try/catch in case auditResults is missing an audit result
+      try {
+        resolvedMetrics.push({
+          id: metric.id,
+          name: metric.name,
+          ts: metric.getTs(auditResults)
+        });
+      } catch (e) {
+        log.error('metrics-evts', `${metric.name} timestamp not found: ${e.message}`);
+      }
+    });
+    return resolvedMetrics;
+  }
+
+  /**
+   * Getter for our navigationStart trace event
+   */
+  get navigationStartEvt() {
+    if (!this._navigationStartEvt) {
+      const filteredEvents = this._traceEvents.filter(e => {
+        return e.name === 'TracingStartedInPage' || e.cat === 'blink.user_timing';
+      });
+
+      const tracingStartedEvt = filteredEvents.filter(e => e.name === 'TracingStartedInPage')[0];
+      const navigationStartEvt = filteredEvents.filter(e => {
+        return e.name === 'navigationStart' &&
+            e.pid === tracingStartedEvt.pid && e.tid === tracingStartedEvt.tid;
+      })[0];
+      this._navigationStartEvt = navigationStartEvt;
+    }
+    return this._navigationStartEvt;
+  }
+
+
+  /**
+   * Constructs performance.measure trace events, which have start/end events as follows:
+   *     { "pid": 89922,"tid":1295,"ts":77176783452,"ph":"b","cat":"blink.user_timing","name":"innermeasure","args":{},"tts":1257886,"id":"0xe66c67"}
+   *     { "pid": 89922,"tid":1295,"ts":77176882592,"ph":"e","cat":"blink.user_timing","name":"innermeasure","args":{},"tts":1257898,"id":"0xe66c67"}
+   * @param  {!Object} metric
+   * @param  {number} navStartTs
+   * @return {Array} Pair of trace events (start/end)
+   */
+  synthesizeEventPair(metric, navStartTs) {
+    if (!metric.ts || metric.id === 'navstart') {
+      return [];
+    }
+
+    // We'll masquerade our fake events to look mostly like navigationStart
+    const eventBase = {
+      pid: this.navigationStartEvt.pid,
+      tid: this.navigationStartEvt.tid,
+      cat: 'blink.user_timing',
+      name: metric.name,
+      args: {},
+      // randomized id is same for the pair
+      id: `0x${((Math.random() * 1000000) | 0).toString(16)}`
+    };
+    const fakeMeasureStartEvent = Object.assign({}, eventBase, {
+      ts: navStartTs,
+      ph: 'b'
+    });
+    const fakeMeasureEndEvent = Object.assign({}, eventBase, {
+      ts: metric.ts,
+      ph: 'e',
+    });
+    return [fakeMeasureStartEvent, fakeMeasureEndEvent];
+  }
+
+  /**
+   * @returns {Array} User timing raw trace event pairs
+   */
+  generateFakeEvents() {
```

Comment:
not sure if you have more plans for this class, but these could all pretty easily be static. Caching  `this.navigationStartEvt` is probably the only real win to having an instance right now, but since you verify that it's equal to the found metric anyway, you could always pass the event into `synthesizeEventPair` instead of just the timestamp